### PR TITLE
[barefoot][sonic-platform] Fix get_system_eeprom_info and refactor eeprom.py

### PR DESCRIPTION
**Pull request by vboykox**
Signed-off-by: Volodymyr Boyko <volodymyrx.boiko@intel.com>

**- Why I did it**
sonic-platform fixups for Barefoot
**- How I did it**

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by vboykox** [comment]
> This pull request **introduces 2 alerts** when merging [66cca3f](https://github.com/Azure/sonic-buildimage/commit/66cca3f46f0c8689b282bdcde1b668a87d443b7b) into [c54a03f](https://github.com/Azure/sonic-buildimage/commit/c54a03fade6c822ea25051882a5745ac5b4fcbf2) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-b2f746968aa96464713c5842c353f87487c52553)
> 
> **new alerts:**
> 
> * 1 for Unused import
> * 1 for Insecure temporary file

i

**Comment by vboykox** [comment]
> This pull request **introduces 2 alerts** when merging [66cca3f](https://github.com/Azure/sonic-buildimage/commit/66cca3f46f0c8689b282bdcde1b668a87d443b7b) into [c54a03f](https://github.com/Azure/sonic-buildimage/commit/c54a03fade6c822ea25051882a5745ac5b4fcbf2) - [view on LGTM.com](https://lgtm.com/projects/g/Azure/sonic-buildimage/rev/pr-b2f746968aa96464713c5842c353f87487c52553)
> 
> **new alerts:**
> 
> * 1 for Unused import
> * 1 for Insecure temporary file

irrelevant for this pr, changes that were causing this are moved into separate pr

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py [hunk 1]
```diff
@@ -122,36 +97,58 @@ def eeprom_parse(self):
...
+        EEPROM_DECODE_HEADLINES = 6
+        lines = decode_output.replace('\0', '').split('\n')
+        lines = lines[EEPROM_DECODE_HEADLINES:]
+        res = dict()
 
-    def serial_str(self):
-        return self.eeprom.prod_ser_num
+        for line in lines:
+            try:
+                match = re.search('(0x[0-9a-fA-F]{2})([\s]+[\S]+[\s]+)([\S]+[\s]*[\S]*)', line)
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py:126]: please update with the description what do we expect to match here

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py [hunk 2]
```diff
@@ -14,15 +15,15 @@
...
+    from sonic_platform_base.sonic_eeprom import eeprom_base
+    from sonic_platform_base.sonic_eeprom import eeprom_tlvinfo
 
     from platform_thrift_client import thrift_try
 except ImportError as e:
     raise ImportError (str(e) + "- required module not found")
 
 
-eeprom_default_dict = {
+_platform_eeprom_map = {
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py:26]: please make it class member

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py [hunk 3]
```diff
@@ -33,36 +34,13 @@
...
-product_dict = { "Montara"   : "Wedge100BF-32X-O-AC-F-BF",
+_product_dict = { "Montara"   : "Wedge100BF-32X-O-AC-F-BF",
                  "Lower MAV" : "Wedge100BF-65X-O-AC-F-BF",
                  "Upper MAV" : "Wedge100BF-65X-O-AC-F-BF"
                }
 
-EEPROM_SYMLINK = "/var/run/platform/eeprom/syseeprom"
-EEPROM_STATUS = "/var/run/platform/eeprom/status"
+_EEPROM_SYMLINK = "/var/run/platform/eeprom/syseeprom"
+_EEPROM_STATUS = "/var/run/platform/eeprom/status"
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py:43]: please make it class member

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py (hunk 1 above)
Comment (vboykox) [reply platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/eeprom.py:127]: done

**Review by akokhan — APPROVED** [review]

**Comment by vboykox** [comment]
@jleveque

**Review by jleveque — APPROVED** [review]

**Pull request merged**